src/spatialdata/_core/query/spatial_query.py

```diff
@@ -49,7 +50,7 @@ def _get_bounding_box_corners_in_intrinsic_coordinates(
     min_coordinate: list[Number] | ArrayLike,
     max_coordinate: list[Number] | ArrayLike,
     target_coordinate_system: str,
-) -> tuple[ArrayLike, tuple[str, ...]]:
+) -> tuple[ArrayLike | DataArray, tuple[str, ...]]:
```

Comment:
to make the code a bit easier I would always return a `DataArray` for all the types of elements, and adjust the code using this function to deal with the returned `DataArray` instead of the returned `ArrayLike`.

Reply:
Done